src/test/java/org/threeten/extra/chrono/TestInternationalFixedChronology.java

```diff
@@ -0,0 +1,1191 @@
+        assertEquals(InternationalFixedChronology.INSTANCE.range(ChronoField.EPOCH_DAY),             ValueRange.of(-719528, 365_000_000));
+    }
+
+    //-----------------------------------------------------------------------
+    // InternationalFixedDate.range
+    //-----------------------------------------------------------------------
+    @DataProvider(name = "ranges")
+    Object[][] data_ranges() {
+        return new Object[][] {
+                { 2011,  1, 23, ChronoField.DAY_OF_MONTH,          1,  28 },
+                { 2012,  1, 23, ChronoField.DAY_OF_MONTH,          1,  28 },
+
+                { 2011,  1,  1, ChronoField.DAY_OF_WEEK,           1,   7 },
+                { 2012,  1,  1, ChronoField.DAY_OF_WEEK,           1,   7 },
+
+                { 2011,  1, 23, ChronoField.DAY_OF_YEAR,           1, 365 },
+                { 2012,  1, 23, ChronoField.DAY_OF_YEAR,           1, 366 },
+
+                { 2011,  1, 23, ChronoField.MONTH_OF_YEAR,         1,  13 },
+                { 2012,  1, 23, ChronoField.MONTH_OF_YEAR,         1,  13 },
+
+                { 2011,  1,  3, ChronoField.ALIGNED_WEEK_OF_MONTH, 1,   4 },
+                { 2011,  1, 10, ChronoField.ALIGNED_WEEK_OF_MONTH, 1,   4 },
+                { 2011,  1, 17, ChronoField.ALIGNED_WEEK_OF_MONTH, 1,   4 },
+                { 2011,  1, 24, ChronoField.ALIGNED_WEEK_OF_MONTH, 1,   4 },
+                { 2012,  1,  3, ChronoField.ALIGNED_WEEK_OF_MONTH, 1,   4 },
+                { 2012,  1, 10, ChronoField.ALIGNED_WEEK_OF_MONTH, 1,   4 },
+                { 2012,  1, 17, ChronoField.ALIGNED_WEEK_OF_MONTH, 1,   4 },
+                { 2012,  1, 24, ChronoField.ALIGNED_WEEK_OF_MONTH, 1,   4 },
+
+                { 2011,  1, 23, ChronoField.PROLEPTIC_MONTH,      13, 12_999_999 },
+                { 2012,  1, 23, ChronoField.PROLEPTIC_MONTH,      13, 12_999_999 },
+
+                { 2011,  1, 23, ChronoField.YEAR_OF_ERA,           1,    999_999 },
+                { 2012,  1, 23, ChronoField.YEAR_OF_ERA,           1,    999_999 },
+
+                { 2011,  1, 23, ChronoField.YEAR,                  1,    999_999 },
+                { 2012,  1, 23, ChronoField.YEAR,                  1,    999_999 },
+
+                { 2011,  1, 23, ChronoField.EPOCH_DAY,      -719_528, 365_000_000 },
+                { 2012,  1, 23, ChronoField.EPOCH_DAY,      -719_528, 365_000_000 },
+        };
+    }
+
+    @Test(dataProvider = "ranges")
+    public void test_range(
+            final int year,
+            final int month,
+            final int dom,
+            final TemporalField field,
+            final int expectedMin,
+            final int expectedMax) {
+        for (int m = month; m < 14; m++) {
+            assertEquals(InternationalFixedDate.of(year, m, dom).range(field), ValueRange.of(expectedMin, expectedMax));
+        }
+    }
+
+    @DataProvider(name = "rangesLeapAndYearDay")
+    Object[][] data_ranges_leap_and_year_day() {
```

Comment:
Once you set up the date to accept flag values for month/day-of-month for leap-day and year-day, you should put these tests with the rest of the range tests.